src/App.tsx

```diff
@@ -17,18 +56,42 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                value={query}
+                onReset={() => setQuery('')}
+                setQuery={setQuery}
+                setFilterType={(filter) => setFilterType(filter)}
+                filterType={filterType}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {
+                todos.length > 0
+                  ? (
+                    <TodoList
+                      todos={visibleTodos}
+                      selectedTodoId={userId}
+                      selectTodo={(id) => setUserId(id)}
+                    />
+                  )
+                  : (
+                    <Loader />
+                  )
+              }
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {
+        selectedTodo && (
+          <TodoModal
+            selectedTodo={selectedTodo}
+            selectTodo={(id) => setUserId(id)}
+          />
+        )
+      }
```

Comment:
Looks better, i think ( up to you )
```suggestion
      {selectedTodo && (
          <TodoModal
            selectedTodo={selectedTodo}
            selectTodo={(id) => setUserId(id)}
          />
        )}
```